kafka/consumer.go

```diff
@@ -329,15 +329,20 @@ func (c *Consumer) rebalance(ev Event) bool {
 // consumerReader reads messages and events from the librdkafka consumer queue
 // and posts them on the consumer channel.
 // Runs until termChan closes
-func consumerReader(c *Consumer, termChan chan bool) {
+func consumerReader(c *Consumer, termChan chan bool, eventsChanSize int) {
 
 out:
 	for true {
 		select {
 		case _ = <-termChan:
 			break out
 		default:
-			_, term := c.handle.eventPoll(c.events, 100, 1000, termChan)
+			_, term := c.handle.eventPoll(
+				c.events,
+				100,
+				eventsChanSize - len(c.events), // max events depends on available buffer
```

Comment:
Well, considering you can configure the channel size to less than 1000, the current call in `consumer.go` (`_, term := c.handle.eventPoll(c.events, 100, 1000, termChan)`) doesn't quite feel right. If I want to buffer 10,000 events, then surely that should be doable in a single call  (rather than 10 calls as is the case now). Similarly, if I set the buffer to 10, it doesn't make sense to have `eventPoll` loop with a max of 1000 events. Sure, either the channel will be full, or there won't be any messages left to consume, but it feels/looks a bit odd to my eye